pkg/services/publicdashboards/database/database.go

```diff
@@ -264,3 +207,62 @@ func (d *PublicDashboardStoreImpl) GetOrgIdByAccessToken(ctx context.Context, ac
 
 	return orgId, err
 }
+
+// Creates a public dashboard
+func (d *PublicDashboardStoreImpl) Create(ctx context.Context, cmd SavePublicDashboardCommand) (int64, error) {
+	if cmd.PublicDashboard.DashboardUid == "" {
+		return 0, dashboards.ErrDashboardIdentifierNotSet
+	}
+
+	var affectedRows int64
+	err := d.sqlStore.WithDbSession(ctx, func(sess *db.Session) error {
+		var err error
+		affectedRows, err = sess.UseBool("is_enabled").Insert(&cmd.PublicDashboard)
+		return err
+	})
+
+	return affectedRows, err
+}
+
+// Updates existing public dashboard
+func (d *PublicDashboardStoreImpl) Update(ctx context.Context, cmd SavePublicDashboardCommand) (int64, error) {
+	var affectedRows int64
+	err := d.sqlStore.WithDbSession(ctx, func(sess *db.Session) error {
+		timeSettingsJSON, err := json.Marshal(cmd.PublicDashboard.TimeSettings)
+		if err != nil {
+			return err
+		}
+
+		sqlResult, err := sess.Exec("UPDATE dashboard_public SET is_enabled = ?, annotations_enabled = ?, time_settings = ?, updated_by = ?, updated_at = ? WHERE uid = ?",
+			cmd.PublicDashboard.IsEnabled,
+			cmd.PublicDashboard.AnnotationsEnabled,
+			string(timeSettingsJSON),
+			cmd.PublicDashboard.UpdatedBy,
+			cmd.PublicDashboard.UpdatedAt.UTC().Format("2006-01-02 15:04:05"),
+			cmd.PublicDashboard.Uid)
+
+		if err != nil {
+			return err
+		}
+
+		affectedRows, err = sqlResult.RowsAffected()
+
+		return nil
```

Comment:
we're not doing anything with err and returning nil without checking
```suggestion
		affectedRows, err = sqlResult.RowsAffected()

		return err
```